container-engine-lib/lib/backend_interface/objects/service/service_config.go

```diff
@@ -190,27 +217,51 @@ func (serviceConfig *ServiceConfig) GetMinCPUAllocationMillicpus() uint64 {
 	return serviceConfig.privateServiceConfig.MinCpuAllocationMilliCpus
 }
 
+func (serviceConfig *ServiceConfig) SetMinCPUAllocationMillicpus(cpuAllocation uint64) {
+	serviceConfig.privateServiceConfig.MemoryAllocationMegabytes = cpuAllocation
+}
+
 // only available for Kubernetes
 func (serviceConfig *ServiceConfig) GetMinMemoryAllocationMegabytes() uint64 {
 	return serviceConfig.privateServiceConfig.MinMemoryAllocationMegabytes
 }
 
+func (serviceConfig *ServiceConfig) SetMinMemoryAllocationMegabytes(memoryAllocation uint64) {
+	serviceConfig.privateServiceConfig.MemoryAllocationMegabytes = memoryAllocation
+}
+
 func (serviceConfig *ServiceConfig) GetUser() *service_user.ServiceUser {
 	return serviceConfig.privateServiceConfig.User
 }
 
+func (serviceConfig *ServiceConfig) SetUser(user *service_user.ServiceUser) {
```

Comment:
with so many setters i wonder whether can just have these variables as not private anymore;

Reply:
Def a good point. What do you think about doing this when we've enabled the ability to set all of the `ServiceConfig` values? I think there's value in keeping `Ports`, `EnvVars`, `Entrypoint`.. any ServiceConfig value affected by future references private until we have a way to handle setting them carefully.